### PR TITLE
r/aws_network_interface - read after create + add support for ipv6

### LINKED ISSUE #1707: add ipv6_ip address to aws_network_interface
Hi there,

Please add ipv6_ip address to aws_network_interface
http://docs.aws.amazon.com/cli/latest/reference/ec2/create-network-interface.html

Thank you.

### LINKED ISSUE #8408: Could not get all ipv6 information(ipv6_addresses) from terraform.tfstate file
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.11
+ provider.aws v2.7.0

### Affected Resource(s)

*aws_network_interface
*aws_instance

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
variable "net-01_ids" {}
resource "aws_network_interface" "net-01" {
  subnet_id = "${element(split(",", var.net-01_ids), var.zone_index)}"
  security_groups = [ "${split(",", var.security_group_ids)}"]
  source_dest_check = false
  tags {
    Name = "${var.tag_name}"
  }
}
variable "net-02_ids" {}
resource "aws_network_interface" "net-02" {
  subnet_id = "${element(split(",", var.net-02_ids), var.zone_index)}"
  security_groups = [ "${split(",", var.security_group_ids)}"]
  source_dest_check = false
  tags {
    Name = "${var.tag_name}"
  }
}

resource "aws_instance" "instance" {
    ami = "${var.source_ami}"
    instance_type = "${var.instance_type}"
    key_name = "${var.ssh_key_pair}" 

    network_interface {
    network_interface_id = "${aws_network_interface.net-01.id}"
    device_index = 0
   } 

    network_interface {
    network_interface_id = "${aws_network_interface.net-02.id}"
    device_index = 1
   } 

    user_data = "${file("${var.user_data_file}")}" 

    root_block_device {
        delete_on_termination = true
        volume_size = "200"
        volume_type = "gp2"
    }

    tags {
        Name = "${var.tag_name}"
        Alias = "${var.tag_name}"
        sshUser = "${var.ssh_username}"
        role = "${var.role}"
        KubernetesCluster = "${var.short_name}"
        "kubernetes.io/cluster/serena" = "${var.short_name}"
    }
}
```

### Expected Behavior
I enabled Auto-assign IPv6 address, and when instance launched there are 2 ipv6 address configured in the instance:
```hcl
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP group default qlen 1000
    link/ether 02:c1:87:29:a3:cc brd ff:ff:ff:ff:ff:ff
    inet 192.16.1.53/24 brd 192.16.1.255 scope global dynamic eth0
       valid_lft 2486sec preferred_lft 2486sec
    inet6 2600:1f16:89a:9e01:4688:1a59:e66:10b4/64 scope global dynamic 
       valid_lft 442sec preferred_lft 142sec
    inet6 fe80::c1:87ff:fe29:a3cc/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP group default qlen 1000
    link/ether 02:ff:a1:d0:44:1c brd ff:ff:ff:ff:ff:ff
    inet 192.16.1.92/24 brd 192.16.1.255 scope global dynamic eth1
       valid_lft 2491sec preferred_lft 2491sec
    inet6 2600:1f16:89a:9e01:589:29e0:8517:84ec/64 scope global dynamic 
       valid_lft 381sec preferred_lft 81sec
    inet6 fe80::ff:a1ff:fed0:441c/64 scope link 
```
**Expected:** Could get ipv6 information like ipv6 address form **'aws_network_interface'** of terraform.tfstate file. 
### Actual Behavior
There are no ipv6 information in **'aws_network_interface'** of terraform.tfstate file. 
```hcl
                "aws_network_interface.net-01": {
                    "type": "aws_network_interface",
                    "depends_on": [],
                    "primary": {
                        "id": "eni-0ac6ec627d24b1f6b",
                        "attributes": {
                            "attachment.#": "0",
                            "description": "",
                            "id": "eni-0ac6ec627d24b1f6b",
                            "private_dns_name": "",
                            "private_ip": "192.16.1.53",
                            "private_ips.#": "1",
                            "private_ips.1071487055": "192.16.1.53",
                            "security_groups.#": "1",
                            "security_groups.791620435": "sg-befcc7d4",
                            "source_dest_check": "false",
                            "subnet_id": "subnet-17b6e97f",
                            "tags.%": "1",
                            "tags.Name": "serena-01"
                        },
                        "meta": {},
                        "tainted": false
                    },
                    "deposed": [],
                    "provider": "provider.aws"
                }
```
Could only get one ipv6 address of eth0 in **'aws_instance.instance'** from terraform.tfstate file. 
```hcl
               "aws_instance.instance": {
                    "type": "aws_instance",
                    "depends_on": [
                        "aws_network_interface.net-01",
                        "aws_network_interface.net-02"
                    ],
                    "primary": {
                        "id": "i-0688d68e26e8b427f",
                        "attributes": {
...
                            "instance_type": "c4.xlarge",
                            "ipv6_address_count": "1",
                            "ipv6_addresses.#": "1",
                            "ipv6_addresses.0": "2600:1f16:89a:9e01:4688:1a59:e66:10b4",

```
### Steps to Reproduce
Right click the subnet with you want to use ipv6, and then click Modify auto-assign IP settings
Select the check box Auto-assign IPv6 and Save.
Then use this subnet as I decribed in **Terraform Configuration Files**

1. `terraform apply`

### LINKED ISSUE #12260: resource aws_network_interface - add ipv6 support
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It seems you can't use `ipv6_addresses` or `ipv6_address_count` with aws_network_interface. Requesting addition of support here. Issue is: if you have an EC2 instance with multiple ENIs, you can't manage ipv6 with it. You can only seem to mange ipv6 with and ENI when it is the primary ENI via aws_instance using arguments `ipv6_addresses` or `ipv6_address_count`. I believe we should have the same two arguments available for aws_network_interface. 

I believe `vendor\github.com\aws\aws-sdk-go\service\ec2\api.go` has the necessary methods to call from `aws\resource_aws_network_interface.go`.

This is regarding provider version 2.51.0.

### New or Affected Resource(s)

* aws_network_interface (affected)

### Potential Terraform Configuration

Example - I'd like to be able to something like this and assign both ipv6 and private ipv4 addresses to an ENI. Today, ipv6_addresses is not a valid argument for this resource (nor is ipv6_address_count).
```hcl
resource "aws_network_interface" "fw1_eni" {
  description       = "interface 1 for firewall 1"
  subnet_id         = var.subnet_ids[0]
  ipv6_addresses = [var.ips_v6[0]]
  private_ips       = [var.ips_v4[0]]
  security_groups   = [aws_security_group.fw_sg_1.id]
  source_dest_check = false
}
```

### References

* My request here looks similar to this one that has been open since 2017: https://github.com/terraform-providers/terraform-provider-aws/issues/1707.

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12260
Closes #1707
Closes #8408
Relates #1060

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_network_interface: add support for adding IPV6 addresses
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSENI_'
--- PASS: TestAccAWSENI_ipv6 (239.85s)
--- PASS: TestAccAWSENI_tags (241.01s)
--- PASS: TestAccAWSENI_ipv6_count (301.36s)
--- PASS: TestAccAWSENI_sourceDestCheck (235.46s)
--- PASS: TestAccAWSENI_computedIPs (117.03s)
--- PASS: TestAccAWSENI_PrivateIpsCount (253.85s)
--- PASS: TestAccAWSENI_basic (122.11s)
--- PASS: TestAccAWSENI_disappears (100.06s)
--- PASS: TestAccAWSENI_updatedDescription (194.35s)
--- PASS: TestAccAWSENI_attached (214.29s)
--- PASS: TestAccAWSENI_ignoreExternalAttachment (173.77s)
```

### CODE REVIEW

File: aws/resource_aws_network_interface.go [hunk 1]
```diff
@@ -108,6 +107,18 @@ func resourceAwsNetworkInterface() *schema.Resource {
 			},
 
 			"tags": tagsSchema(),
+			"ipv6_address_count": {
```
Comment (ewbankkit) [inline aws/resource_aws_network_interface.go]: I think `ipv6_address_count` conflicts with `ipv6_addresses`.

**Comment by DrFaust92** [comment]
@bflad, i have seem to have removed `terraform-providers/aws-provider` from the reviewers. can you assist in fixing this?

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSENI_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 4 -run=TestAccAWSENI_ -timeout 120m
=== RUN   TestAccAWSENI_basic
=== PAUSE TestAccAWSENI_basic
=== RUN   TestAccAWSENI_disappears
=== PAUSE TestAccAWSENI_disappears
=== RUN   TestAccAWSENI_updatedDescription
=== PAUSE TestAccAWSENI_updatedDescription
=== RUN   TestAccAWSENI_attached
=== PAUSE TestAccAWSENI_attached
=== RUN   TestAccAWSENI_ignoreExternalAttachment
=== PAUSE TestAccAWSENI_ignoreExternalAttachment
=== RUN   TestAccAWSENI_sourceDestCheck
=== PAUSE TestAccAWSENI_sourceDestCheck
=== RUN   TestAccAWSENI_computedIPs
=== PAUSE TestAccAWSENI_computedIPs
=== RUN   TestAccAWSENI_PrivateIpsCount
=== PAUSE TestAccAWSENI_PrivateIpsCount
=== CONT  TestAccAWSENI_basic
=== CONT  TestAccAWSENI_sourceDestCheck
=== CONT  TestAccAWSENI_PrivateIpsCount
=== CONT  TestAccAWSENI_computedIPs
--- PASS: TestAccAWSENI_computedIPs (42.16s)
=== CONT  TestAccAWSENI_ignoreExternalAttachment
--- PASS: TestAccAWSENI_sourceDestCheck (42.68s)
=== CONT  TestAccAWSENI_attached
--- PASS: TestAccAWSENI_basic (45.69s)
=== CONT  TestAccAWSENI_updatedDescription
--- PASS: TestAccAWSENI_updatedDescription (71.77s)
=== CONT  TestAccAWSENI_disappears
--- PASS: TestAccAWSENI_PrivateIpsCount (119.91s)
--- PASS: TestAccAWSENI_ignoreExternalAttachment (112.27s)
--- PASS: TestAccAWSENI_disappears (40.67s)
--- PASS: TestAccAWSENI_attached (237.41s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	280.148s
```

**Comment by euank** [comment]
I'm not a valid reviewer in this repo, but for what it's worth, this change LGTM, both code-wise and from using it. I've patched this into the plugin locally, and it's worked well for managing ipv6 addresses on my ENIs.

**Comment by DrFaust92** [comment]
Rebased

File: aws/resource_aws_network_interface_test.go [hunk 2]
```diff
@@ -486,248 +607,285 @@ func testAccCheckAWSENIMakeExternalAttachment(n string, conf *ec2.NetworkInterfa
...
+  subnet_id         = "${aws_subnet.test.id}"
+  source_dest_check = false
 }
 `
 
 const testAccAWSENIConfigWithAttachment = `
-resource "aws_vpc" "foo" {
-	cidr_block = "172.16.0.0/16"
-	enable_dns_hostnames = true
+data "aws_availability_zones" "available" {
```
Comment (ewbankkit) [inline aws/resource_aws_network_interface_test.go:766]: Change to

```hcl
data "aws_availability_zones" "available" {
  state = "available"
 
  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}
```

(see #12412, #12517) else I get the error

```
--- FAIL: TestAccAWSENI_ignoreExternalAttachment (62.85s)
    testing.go:669: Step 0 error: errors during apply:
        
        Error: Error launching source instance: Unsupported: The requested configuration is currently not supported. Please check the documentation for supported configurations.
        	status code: 400, request id: b71e5cb8-fcca-44ae-85ee-8a976eb52998
        
          on /tmp/tf-test739947919/main.tf line 41:
          (source code not available)
```

File: aws/resource_aws_network_interface_test.go [hunk 3]
```diff
@@ -486,248 +607,285 @@ func testAccCheckAWSENIMakeExternalAttachment(n string, conf *ec2.NetworkInterfa
...
 
-const testAccAWSENIConfig = `
-resource "aws_vpc" "foo" {
-	cidr_block = "172.16.0.0/16"
-	enable_dns_hostnames = true
-	tags = {
-		Name = "terraform-testacc-network-interface"
-	}
+const testAccAWSENIConfigBase = `
+data "aws_availability_zones" "available" {
```
Comment (ewbankkit) [inline aws/resource_aws_network_interface_test.go:611]: Change to

```hcl
data "aws_availability_zones" "available" {
  state = "available"
 
  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}
```

(see #12412, #12517), else I get the error

```
--- FAIL: TestAccAWSENI_basic (62.89s)
    testing.go:669: Step 0 error: Check failed: Check 2/8 error: expected availability_zone to be us-west-2a, but was us-west-2-lax-1a
```

**Comment by DrFaust92** [comment]
@ewbankkit, fixed!

**Comment by ewbankkit** [comment]
```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSENI_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSENI_ -timeout 120m
=== RUN   TestAccAWSENI_basic
=== PAUSE TestAccAWSENI_basic
=== RUN   TestAccAWSENI_ipv6
=== PAUSE TestAccAWSENI_ipv6
=== RUN   TestAccAWSENI_ipv6_count
=== PAUSE TestAccAWSENI_ipv6_count
=== RUN   TestAccAWSENI_disappears
=== PAUSE TestAccAWSENI_disappears
=== RUN   TestAccAWSENI_updatedDescription
=== PAUSE TestAccAWSENI_updatedDescription
=== RUN   TestAccAWSENI_attached
=== PAUSE TestAccAWSENI_attached
=== RUN   TestAccAWSENI_ignoreExternalAttachment
=== PAUSE TestAccAWSENI_ignoreExternalAttachment
=== RUN   TestAccAWSENI_sourceDestCheck
=== PAUSE TestAccAWSENI_sourceDestCheck
=== RUN   TestAccAWSENI_computedIPs
=== PAUSE TestAccAWSENI_computedIPs
=== RUN   TestAccAWSENI_PrivateIpsCount
=== PAUSE TestAccAWSENI_PrivateIpsCount
=== RUN   TestAccAWSENI_tags
=== PAUSE TestAccAWSENI_tags
=== CONT  TestAccAWSENI_basic
=== CONT  TestAccAWSENI_tags
--- PASS: TestAccAWSENI_basic (76.21s)
=== CONT  TestAccAWSENI_PrivateIpsCount
--- PASS: TestAccAWSENI_tags (127.02s)
=== CONT  TestAccAWSENI_computedIPs
--- PASS: TestAccAWSENI_computedIPs (74.49s)
=== CONT  TestAccAWSENI_sourceDestCheck
--- PASS: TestAccAWSENI_PrivateIpsCount (152.00s)
=== CONT  TestAccAWSENI_ignoreExternalAttachment
--- PASS: TestAccAWSENI_sourceDestCheck (127.38s)
=== CONT  TestAccAWSENI_attached
--- PASS: TestAccAWSENI_ignoreExternalAttachment (115.07s)
=== CONT  TestAccAWSENI_updatedDescription
--- FAIL: TestAccAWSENI_attached (28.63s)
    testing.go:684: Step 0 error: errors during apply:
        
        Error: Error launching source instance: Unsupported: The requested configuration is currently not supported. Please check the documentation for supported configurations.
        	status code: 400, request id: 05f5ab80-b6b3-4354-8874-b0d3620e7f18
        
          on /tmp/tf-test893588910/main.tf line 39:
          (source code not available)
        
        
=== CONT  TestAccAWSENI_disappears
--- PASS: TestAccAWSENI_disappears (74.96s)
=== CONT  TestAccAWSENI_ipv6_count
--- PASS: TestAccAWSENI_updatedDescription (102.77s)
=== CONT  TestAccAWSENI_ipv6
--- PASS: TestAccAWSENI_ipv6 (76.82s)
--- PASS: TestAccAWSENI_ipv6_count (151.75s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	584.297s
FAIL
GNUmakefile:26: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

It looks like we need to add

```hcl
data "aws_availability_zones" "available" {
  state = "available"
 
  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}
```

to `TestAccAWSENI_attached`.

**Comment by DrFaust92** [comment]
fixed again

**Comment by ewbankkit** [comment]
```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSENI_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSENI_ -timeout 120m
=== RUN   TestAccAWSENI_basic
=== PAUSE TestAccAWSENI_basic
=== RUN   TestAccAWSENI_ipv6
=== PAUSE TestAccAWSENI_ipv6
=== RUN   TestAccAWSENI_ipv6_count
=== PAUSE TestAccAWSENI_ipv6_count
=== RUN   TestAccAWSENI_disappears
=== PAUSE TestAccAWSENI_disappears
=== RUN   TestAccAWSENI_updatedDescription
=== PAUSE TestAccAWSENI_updatedDescription
=== RUN   TestAccAWSENI_attached
=== PAUSE TestAccAWSENI_attached
=== RUN   TestAccAWSENI_ignoreExternalAttachment
=== PAUSE TestAccAWSENI_ignoreExternalAttachment
=== RUN   TestAccAWSENI_sourceDestCheck
=== PAUSE TestAccAWSENI_sourceDestCheck
=== RUN   TestAccAWSENI_computedIPs
=== PAUSE TestAccAWSENI_computedIPs
=== RUN   TestAccAWSENI_PrivateIpsCount
=== PAUSE TestAccAWSENI_PrivateIpsCount
=== RUN   TestAccAWSENI_tags
=== PAUSE TestAccAWSENI_tags
=== CONT  TestAccAWSENI_basic
=== CONT  TestAccAWSENI_ignoreExternalAttachment
--- PASS: TestAccAWSENI_basic (77.45s)
=== CONT  TestAccAWSENI_tags
--- PASS: TestAccAWSENI_ignoreExternalAttachment (114.93s)
=== CONT  TestAccAWSENI_PrivateIpsCount
--- PASS: TestAccAWSENI_tags (126.79s)
=== CONT  TestAccAWSENI_computedIPs
--- PASS: TestAccAWSENI_PrivateIpsCount (151.92s)
=== CONT  TestAccAWSENI_sourceDestCheck
--- PASS: TestAccAWSENI_computedIPs (75.15s)
=== CONT  TestAccAWSENI_disappears
--- PASS: TestAccAWSENI_disappears (74.97s)
=== CONT  TestAccAWSENI_attached
--- PASS: TestAccAWSENI_sourceDestCheck (126.19s)
=== CONT  TestAccAWSENI_updatedDescription
--- PASS: TestAccAWSENI_updatedDescription (103.28s)
=== CONT  TestAccAWSENI_ipv6_count
--- PASS: TestAccAWSENI_attached (271.79s)
=== CONT  TestAccAWSENI_ipv6
--- PASS: TestAccAWSENI_ipv6_count (153.18s)
--- PASS: TestAccAWSENI_ipv6 (77.86s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	704.046s
```

**Comment by DrFaust92** [comment]
@ewbankkit, as always thanks a lot!

**Comment by bryceml** [comment]
Just curious if there is something holding this up from being merged, I'd like to be able to use this.  Last comment was about 6 weeks ago.

**Comment by DrFaust92** [comment]
Rebased and Refactored:

```
--- PASS: TestAccAWSENI_ipv6 (239.85s)
--- PASS: TestAccAWSENI_tags (241.01s)
--- PASS: TestAccAWSENI_ipv6_count (301.36s)
--- PASS: TestAccAWSENI_sourceDestCheck (235.46s)
--- PASS: TestAccAWSENI_computedIPs (117.03s)
--- PASS: TestAccAWSENI_PrivateIpsCount (253.85s)
--- PASS: TestAccAWSENI_basic (122.11s)
--- PASS: TestAccAWSENI_disappears (100.06s)
--- PASS: TestAccAWSENI_updatedDescription (194.35s)
--- PASS: TestAccAWSENI_attached (214.29s)
--- PASS: TestAccAWSENI_ignoreExternalAttachment (173.77s)
```

File: aws/resource_aws_network_interface_sg_attachment_test.go [hunk 4]
```diff
@@ -56,7 +56,7 @@ func TestAccAWSNetworkInterfaceSGAttachment_disappears(t *testing.T) {
 				Config: testAccAwsNetworkInterfaceSGAttachmentConfig(rName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAWSNetworkInterfaceSGAttachmentExists(resourceName, &networkInterface),
-					testAccCheckAWSENIDisappears(&networkInterface),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsNetworkInterface(), resourceName),
```
Comment (bflad) [inline aws/resource_aws_network_interface_sg_attachment_test.go:59]: This test has a new failure:

```
=== CONT  TestAccAWSNetworkInterfaceSGAttachment_disappears
TestAccAWSNetworkInterfaceSGAttachment_disappears: resource_aws_network_interface_sg_attachment_test.go:42: [INFO] Got non-empty plan, as expected
TestAccAWSNetworkInterfaceSGAttachment_disappears: resource_aws_network_interface_sg_attachment_test.go:42: Check 2/2 error: Error deleting ENI: InvalidNetworkInterfaceId.Malformed: Invalid id: "sg-0beb8cdaa5e27fcd2_eni-010a1844f2ddecf3d" (expecting "eni-...")
  status code: 400, request id: e0f2e0d8-d7a8-4096-8dcf-579ac880197c
--- FAIL: TestAccAWSNetworkInterfaceSGAttachment_disappears (87.79s)
```

The resource needs to align with the type:

```suggestion
					testAccCheckResourceDisappears(testAccProvider, resourceAwsNetworkInterfaceSGAttachment(), resourceName),
```

Comment (DrFaust92) [reply aws/resource_aws_network_interface_sg_attachment_test.go]: --- PASS: TestAccAWSNetworkInterfaceSGAttachment_disappears (186.56s)
Thanks!

**Comment by DrFaust92** [comment]
Rebased + more test refactor:

```
--- PASS: TestAccAWSENI_computedIPs (114.66s)
--- PASS: TestAccAWSENI_basic (121.19s)
--- PASS: TestAccAWSENI_attached (194.78s)
--- PASS: TestAccAWSENI_PrivateIpsCount (259.21s)
--- PASS: TestAccAWSENI_ignoreExternalAttachment (182.95s)
--- PASS: TestAccAWSENI_sourceDestCheck (238.15s)
--- PASS: TestAccAWSENI_disappears (99.56s)
--- PASS: TestAccAWSENI_updatedDescription (191.10s)
--- PASS: TestAccAWSENI_ipv6_count (298.87s)
--- PASS: TestAccAWSENI_tags (238.05s)
--- PASS: TestAccAWSENI_ipv6 (248.03s)
```

**Review by bflad — APPROVED** [review]
Looks great, thanks @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSENI_attached (183.12s)
--- PASS: TestAccAWSENI_basic (102.76s)
--- PASS: TestAccAWSENI_computedIPs (101.72s)
--- PASS: TestAccAWSENI_disappears (87.26s)
--- PASS: TestAccAWSENI_ignoreExternalAttachment (185.44s)
--- PASS: TestAccAWSENI_ipv6 (141.80s)
--- PASS: TestAccAWSENI_ipv6_count (159.75s)
--- PASS: TestAccAWSENI_PrivateIpsCount (151.93s)
--- PASS: TestAccAWSENI_sourceDestCheck (143.44s)
--- PASS: TestAccAWSENI_tags (145.50s)
--- PASS: TestAccAWSENI_updatedDescription (123.99s)

--- PASS: TestAccAWSNetworkInterfaceSGAttachment_basic (70.59s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_DataSource (119.79s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_disappears (108.71s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_Instance (139.32s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_Multiple (90.30s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSENI_attached (178.91s)
--- PASS: TestAccAWSENI_basic (95.41s)
--- PASS: TestAccAWSENI_computedIPs (93.75s)
--- PASS: TestAccAWSENI_disappears (85.16s)
--- PASS: TestAccAWSENI_ignoreExternalAttachment (146.43s)
--- PASS: TestAccAWSENI_ipv6 (131.01s)
--- PASS: TestAccAWSENI_ipv6_count (141.75s)
--- PASS: TestAccAWSENI_PrivateIpsCount (85.91s)
--- PASS: TestAccAWSENI_sourceDestCheck (131.61s)
--- PASS: TestAccAWSENI_tags (130.60s)
--- PASS: TestAccAWSENI_updatedDescription (112.40s)

--- PASS: TestAccAWSNetworkInterfaceSGAttachment_basic (68.81s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_DataSource (86.84s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_disappears (102.20s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_Instance (121.17s)
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_Multiple (83.78s)
```

**Pull request merged**